trin-core/src/types/accumulator.rs

```diff
@@ -149,6 +147,67 @@ impl MasterAccumulator {
     }
 }
 
+/// Validation function for pre merge headers that will use the master accumulator to validate
+/// whether or not the given header is canonical. Returns true if header is validated, false if
+/// header is invalidated, or an err if it is unable to validate the header using the master
+/// accumulator.
+pub async fn validate_pre_merge_header(
+    header: &Header,
+    master_acc: &MasterAccumulator,
+    history_jsonrpc_tx: mpsc::UnboundedSender<HistoryJsonRpcRequest>,
+) -> anyhow::Result<bool> {
+    if header.number > MERGE_BLOCK_NUMBER {
+        return Err(anyhow!("Unable to validate post-merge block."));
+    }
+    if header.number >= master_acc.next_header_height() {
```

Comment:
```suggestion
    if header.number > master_acc.height() {
```

Reply:
The problem is that `height()` returns an `Option` which makes this evaluation unnecessarily messy